internal/rbd/controllerserver.go

```diff
@@ -995,6 +1015,17 @@ func (cs *ControllerServer) CreateSnapshot(
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	// Set snapshot-name/snapshot-namespace/snapshotcontent-name details
+	// on RBD backend image as metadata on create
+	if cs.SetMetadata {
+		// Update the metadata on snapshot not on the original image
+		rbdVol.RbdImageName = rbdSnap.RbdSnapName
+		err = rbdVol.setSnapShotMetadata(ctx, req.GetParameters())
```

Comment:
During rbd clone operation all the clones will get the metadata from the parent we need to unset/remove the metadata in below cases

* During CreateSnapshot ->Unset the PVC metadata on the rbd image created for the snapshot
* During CreateVolume From snapshot -> Unset the snapshot metadata from the rbd image created from the snapshot
* During CreateVolume From Volume -> Unset the parent PVC metadata on the temp clone rbd image. 

If this looks too much open up an issue to track we can fix this in followup PR's.